salt/_pillar/metalk8s.py

```diff
@@ -49,12 +51,42 @@ def _load_ca(config_data):
     }
 
 
+def _get_endpoints(config="/etc/kubernetes/admin.conf",
```

Comment:
I'd like to keep pillars-that-dont-need-Kubernetes-available and pillars-that-need-Kubernetes-available separate (in separate modules), so this shouldn't go in here.

Reply:
Do you mean move it in `metalk8s_nodes.py` or create another module? BTW I just see that in `metalk8s_nodes.py` we use directly the `python-kubernetes` python client. In this PR context, is it preferable to use  `python-kubernetes` client directly or do we want to use salt client and then maintain `salt/_modules/kubernetesmod.py`?

Reply:
I'd say, add another pillar module.

You can keep using the `__salt__` procedures, that's fine. The other one uses `python-kubernetes` directly since it's based on some work I did a long time ago. So is the roster module, actually.

I guess over time we should harmonize those.